pkg/webhook/validate.go

```diff
@@ -272,15 +277,15 @@ func validatePolicyForNodePolicy(
 	return true, nil
 }
 
-func validateNicModel(selector *sriovnetworkv1.SriovNetworkNicSelector, iface *sriovnetworkv1.InterfaceExt, node *corev1.Node) bool {
+func validateNicModel(selector *sriovnetworkv1.SriovNetworkNicSelector, iface *sriovnetworkv1.InterfaceExt, node *corev1.Node) error {
 	if selector.Vendor != "" && selector.Vendor != iface.Vendor {
-		return false
+		return fmt.Errorf("selector vendor: %s is not equal to the interface vendor: %s", selector.Vendor, iface.Vendor)
```

Comment:
I don't think we should be considering all these cases as errors. Rather these cases are matching criteria. For example if an Intel NIC being checked does not equal a Nvidia NIC, then it is considered verbose logging to debug an issue. To see which NICs didn't match with the policy.